### ISSUE TITLE
pop comment from dataframe parameter

### ISSUE THREAD

**Issue by rheinheimer** [pywr/pywr#787]
I like parameter comments, but encountered an error when using "comment" with a dataframe parameter. I guess this is because dataframe parameters allow passing additional arbitrary keyword arguments to pandas. The error kicked back is "Only length-1 comment characters supported".

The parameter comment should be popped, though I'm not sure if this should be for parameters generally (i.e., in [load_parameter](https://github.com/pywr/pywr/blob/e821da240cf52df1baa4e10404af335d6e34cde5/pywr/parameters/_parameters.pyx#L1557)) or for dataframes specifically (i.e. in [load_dataframe](https://github.com/pywr/pywr/blob/2132b6e405d653aa56fe69ad860fc5e79778b92c/pywr/dataframe_tools.py#L155)). I'm guessing the latter. Either way, the comment should be retained for the parameter as a whole, just not passed to Pandas.

**Comment by jetuk** [comment]
Thanks for spotting this. I actually do not like the open ended keys being passed to the pandas functions as we currently have it. My preferred solution to this is to move all those keywords in to a separate key dedicated to pandas keywords. Something like this:

```json
{
  "type": "dataframe",
  "url": "mydata.csv",
  "pandas_kwargs": {
    "parse_dates": true,
    "index_col": 0
  }
}
```

This makes the top-level schema a bit more concrete. This has been done in the `schema2` branch, but probably ought to be done in here too. The main issue with this change is that it would break existing JSON format. And there are probably other JSON format / schema refinements that should be done elsewhere. It would be good to do all these changes in one go for a new release. e.g. v2.0.0.

@Batch21 any thoughts on this?

**Comment by rheinheimer** [comment]
That makes sense, and is the approach of some Reach libraries I've used. In the meantime, I just did #788, so an easy fix.

**Comment by jetuk** [comment]
Tagging #641 #611 

Closed in #788
